execute/plugin_usdc_test.go

```diff
@@ -0,0 +1,515 @@
+	require.NoError(t, err)
+	sequenceNumbers = slicelib.Map(outcome.Report.ChainReports[0].Messages, func(m cciptypes.Message) cciptypes.SeqNum {
+		return m.Header.SequenceNumber
+	})
+	require.ElementsMatch(t, sequenceNumbers, []cciptypes.SeqNum{102, 103, 104, 105})
+	//Attestation data added to the both USDC messages
+	require.NotEmpty(t, outcome.Report.ChainReports[0].OffchainTokenData[2])
+	require.NotEmpty(t, outcome.Report.ChainReports[0].OffchainTokenData[3])
+}
+
+type nodeSetup struct {
+	node            *execute.Plugin
+	reportCodec     cciptypes.ExecutePluginCodec
+	msgHasher       cciptypes.MessageHasher
+	TokenDataReader *mock_types.MockTokenDataReader
+}
+
+func setupHomeChainPoller(
+	t *testing.T,
+	donID plugintypes.DonID,
+	lggr logger.Logger,
+	chainConfigInfos []reader.ChainConfigInfo,
+) reader.HomeChain {
+	const ccipConfigAddress = "0xCCIPConfigFakeAddress"
+
+	homeChainReader := readermock.NewMockContractReaderFacade(t)
+	var firstCall = true
+	homeChainReader.On(
+		"GetLatestValue",
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+		mock.MatchedBy(func(input map[string]interface{}) bool {
+			_, pageIndexExists := input["pageIndex"]
+			_, pageSizeExists := input["pageSize"]
+			return pageIndexExists && pageSizeExists
+		}),
+		mock.Anything,
+	).Run(
+		func(args mock.Arguments) {
+			arg := args.Get(4).(*[]reader.ChainConfigInfo)
+			if firstCall {
+				*arg = chainConfigInfos
+				firstCall = false
+			} else {
+				*arg = []reader.ChainConfigInfo{} // return empty for other pages
+			}
+		}).Return(nil)
+
+	homeChainReader.EXPECT().
+		GetLatestValue(mock.Anything, types.BoundContract{
+			Address: ccipConfigAddress,
+			Name:    consts.ContractNameCCIPConfig,
+		}.ReadIdentifier(consts.MethodNameGetOCRConfig), primitives.Unconfirmed, map[string]any{
+			"donId":      donID,
+			"pluginType": consts.PluginTypeExecute,
+		}, mock.Anything).
+		Run(
+			func(
+				ctx context.Context,
+				readIdentifier string,
+				confidenceLevel primitives.ConfidenceLevel,
+				params,
+				returnVal interface{},
+			) {
+				*returnVal.(*[]reader.OCR3ConfigWithMeta) = []reader.OCR3ConfigWithMeta{{}}
+			}).
+		Return(nil)
+
+	homeChain := reader.NewHomeChainConfigPoller(
+		homeChainReader,
+		lggr,
+		// to prevent linting error because of logging after finishing tests, we close the poller after each test, having
+		// lower polling interval make it catch up faster
+		time.Minute,
+		types.BoundContract{
+			Address: ccipConfigAddress,
+			Name:    consts.ContractNameCCIPConfig,
+		},
+	)
+
+	return homeChain
+}
+
+func makeMsg(seqNum cciptypes.SeqNum, src, dest cciptypes.ChainSelector, executed bool) inmem.MessagesWithMetadata {
+	return inmem.MessagesWithMetadata{
+		Message: cciptypes.Message{
+			Header: cciptypes.RampMessageHeader{
+				SourceChainSelector: src,
+				SequenceNumber:      seqNum,
+			},
+		},
+		Destination: dest,
+		Executed:    executed,
+	}
+}
+
+func setupSimpleTest(
```

Comment:
Small description of test scenario would be helpful, e.g "This test sets up 6 messages, 2 of which are already executed, and 3 of which have USDC attestions, 2 of which are not ready immediately", etc.